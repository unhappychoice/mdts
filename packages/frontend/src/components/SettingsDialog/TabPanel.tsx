import React from 'react';
import { Box } from '@mui/material';

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function TabPanel(props: TabPanelProps): React.ReactElement {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`vertical-tabpanel-${index}`}
      aria-labelledby={`vertical-tab-${index}`}
      {...other}
      style={{ width: '100%' }}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          {children}
        </Box>
      )}
    </div>
  );
}

export default TabPanel;
