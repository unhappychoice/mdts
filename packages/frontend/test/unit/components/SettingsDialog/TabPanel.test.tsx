import { render } from '@testing-library/react';
import React from 'react';
import TabPanel from '../../../../src/components/SettingsDialog/TabPanel';

describe('TabPanel', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<TabPanel value={0} index={0}>Test Content</TabPanel>);
    expect(asFragment()).toMatchSnapshot();
  });
});
