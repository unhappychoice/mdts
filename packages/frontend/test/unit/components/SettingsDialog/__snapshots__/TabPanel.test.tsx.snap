// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TabPanel should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="vertical-tab-0"
    id="vertical-tabpanel-0"
    role="tabpanel"
    style="width: 100%;"
  >
    <div
      class="MuiBox-root css-19kzrtu"
    >
      Test Content
    </div>
  </div>
</DocumentFragment>
`;
